File: code/stable/projectile/Projectile_S_NoL_U_U_V/src/cpp/InputFormat.cpp
#include "InputFormat.hpp"

#include <algorithm>
#include <fstream>
#include <limits>
#include <string>

using std::ifstream;
using std::string;

void get_input(string filename, double &v_launch, double &theta, double &p_target) {
    ifstream infile;
    infile.open(filename, std::fstream::in);
    infile.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
    infile >> v_launch;
    infile.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
    infile.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
    infile >> theta;
    infile.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
    infile.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
    infile >> p_target;
    infile.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
    infile.close();
}

File: code/stable/projectile/Projectile_S_NoL_U_U_V/src/csharp/InputFormat.cs
/** \file InputFormat.cs
    \author Samuel J. Crawford, Brooks MacLachlan, and W. Spencer Smith
    \brief Provides the function for reading inputs
*/
using System;
using System.IO;

public class InputFormat {
    
    /** \brief Reads input from a file with the given file name
        \param filename name of the input file
        \param v_launch launch speed: the initial speed of the projectile when launched (m/s)
        \param theta launch angle: the angle between the launcher and a straight line from the launcher to the target (rad)
        \param p_target target position: the distance from the launcher to the target (m)
    */
    public static void get_input(string filename, out double v_launch, out double theta, out double p_target) {
        StreamReader infile;
        infile = new StreamReader(filename);
        infile.ReadLine();
        v_launch = Double.Parse(infile.ReadLine());
        infile.ReadLine();
        theta = Double.Parse(infile.ReadLine());
        infile.ReadLine();
        p_target = Double.Parse(infile.ReadLine());
        infile.Close();
    }
}
